### PR TITLE
Added the FileImageFormatter and the FileSizeFormatter

### PR DESCRIPTION
**Pull request by edurenye**
Explore using formatters on the uri base field to "view files" with configurable formatters.
Added the FileImageFormatter and the FileSizeFormatter.

### CODE REVIEW

File: src/Plugin/Field/FieldFormatter/FileImageFormatter.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      );
+    }
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEntitiesToView(EntityReferenceFieldItemListInterface $items) {
```
Comment (Berdir) [inline src/Plugin/Field/FieldFormatter/FileImageFormatter.php]: you can remove this.

File: src/Plugin/Field/FieldFormatter/FileImageFormatter.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * )
+ */
+class FileImageFormatter extends ImageFormatter {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items) {
+    $elements = array();
+    $files = [$items->getEntity()];;
```
Comment (Berdir) [inline src/Plugin/Field/FieldFormatter/FileImageFormatter.php]: I think we won't be able to use the parent implementation of this, so lets try to simplify it. Drop $files, there's just one, hardcode $delta 0, no empty check, no loop, just $file = $item->getEntity() and use that directly..

File: src/Plugin/Field/FieldFormatter/FileSizeFormatter.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    // This formatter is only available for files.
+    return $field_definition->getTargetEntityTypeId() == 'file';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items) {
+    $elements = array();
+    $files = [$items->getEntity()];;
```
Comment (LKS90) [inline src/Plugin/Field/FieldFormatter/FileSizeFormatter.php]: `array()` on one line, `[]` on the next, please be consistent (IMO [] is nicer than array()). Oh, and two ;; at the end of the line?

**Comment by LKS90** [comment]
Can you repeat that for FileImageFormatter? The FileSizeFormttter uses `[]` and the FileImageFormatter `array()`?

**Comment by LKS90** [comment]
I'd say fix the test fails, but better wait for Sascha's verdict whether the test fails are remotely related to this issue.

**Comment by edurenye** [comment]
It's an unrelated fail, I already discused this with him, so let's move it to needs review for him to take a look.

**Pull request merged**